aten/src/ATen/native/cuda/Blas.cpp

```diff
@@ -819,9 +819,97 @@ static bool _scaled_mm_allowed_device() {
 #endif
 }
 
-// Computes matrix multiply + bias while applying scaling to input and output matrices and computes amax
+namespace{
+
+enum class ScalingType {
+  TensorWise,
+  RowWise,
+  Error
+};
+/*
+ * Scaling Type Determination:
+ * ---------------------------
+ * Conditions and corresponding Scaling Types:
+ *
+ * - If scale_a.numel() == 1 && scale_b.numel() == 1:
+ *   - Returns TensorWise.
+ *
+ * - Else if scale_a.dim() == 1 && scale_a.size(0) == dim_m && scale_b.size(0) == dim_n:
+ *   - Returns RowWise.
+ *
+ * - Otherwise:
+ *   - Returns Error.
+ */
+
+// Validates the scale tensors to scaled_mm
+// And returns the type of scaling/which kernel to use
+ScalingType get_scaling_type(
```

Comment:
I may be out of the loop here, but where exactly is this called?

Reply:
This is not called in this class. When I do cherry-pick, this part of the code is added. There are file differences: https://github.com/pytorch/pytorch/pull/135421/commits/2df620d2e451e8a67f96642033f9eb7ffb099c58#diff-e8a569efee1e650172f120a0fdcda024fe3e4703a4ee3336425c8f685af6b3ab. I need to merge it properly.